src/machine.rs

```diff
@@ -1036,21 +1039,23 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for MiriMachine<'mir, 'tcx> {
         (alloc_id, prov_extra): (AllocId, Self::ProvenanceExtra),
         range: AllocRange,
     ) -> InterpResult<'tcx> {
-        if let Some(data_race) = &mut alloc_extra.data_race {
-            data_race.write(
+        let mut current_span = machine.current_span();
+        if let Some(stacked_borrows) = &mut alloc_extra.stacked_borrows {
```

Comment:
You swapped the order in which the subsystems are called. Why?

Reply:
This _might_ need to happen for the other PR, but honestly it's in here because I was sloppy when separating out this PR.